File: Solutions/0169/0169.kt
// Author: washing
// DateTime: 2021/1/26 18:43
// File: 0169.kt
// Desc: 

class Solution {
    fun majorityElement(nums: IntArray): Int {
        var c = 0
        var a = 0
        for (n in nums){
            if (c == 0){
                a = n
                c += 1
            }else if(a == n){
                c += 1
            }else {
                c -= 1
            }
        }
        return a
    }
}

File: Solutions/0169/0169.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2021/1/26 18:44
# File: 0169.py
# Desc: 

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        c = 0
        for i in nums:
            if not c:
                a = i
                c += 1
            elif i == a:
                c += 1
            else:
                c -= 1
        return a